### PR TITLE
fix: dv handler and common model subscriber fixes

### PR DESCRIPTION
**Pull request by osher-sade**

### CODE REVIEW

File: packages/polaris-typeorm/test/integration-tests/postgres-tests/polaris-repository.test.ts [hunk 1]
```diff
@@ -372,4 +373,22 @@ describe('entity manager tests', () => {
...
     });
+    describe('update entity with cascade save', () => {
+        it('save entity and child entity, child entity is saved with proper data version', async () => {
+            const book: Book | undefined = new Book('foobar');
+            const book2: Book | undefined = new Book('barfoo');
+            const author = new Author('foo', [book, book2]);
+            authorRepo = connection.getRepository(Author, generateContext({ realityId: 1 }));
+            await authorRepo.save(author);
+
+            author.name = 'newName';
```
Comment (ronkatz96) [inline packages/polaris-typeorm/test/integration-tests/postgres-tests/polaris-repository.test.ts:384]: change to foo

File: packages/polaris-typeorm/src/subscribers/common-model-subscriber.ts [hunk 2]
```diff
@@ -21,6 +22,25 @@ export class CommonModelSubscriber implements EntitySubscriberInterface<CommonMo
...
+
+    beforeUpdate(event: UpdateEvent<CommonModel>): Promise<any> | void {
+        const context = (event.manager as PolarisEntityManager).context;
+        const entity = event.entity;
+        if (context && entity instanceof CommonModel) {
+            entity.setDataVersion(context.returnedExtensions?.dataVersion);
+            CommonModelSubscriber.setUpnOfEntity(entity, context);
+        }
+    }
+
```
Comment (itaykl) [inline packages/polaris-typeorm/src/subscribers/common-model-subscriber.ts:43]: unify

File: packages/polaris-typeorm/src/subscribers/common-model-subscriber.ts [hunk 3]
```diff
@@ -114,4 +134,14 @@ export class CommonModelSubscriber implements EntitySubscriberInterface<CommonMo
             CommonModelSubscriber.ncHandler.initNotificationCenterConfigurations(logger, ncConfig);
         }
     }
+
+    private static setUpnOfEntity(entity: CommonModel, context: PolarisGraphQLContext) {
+        if (context?.requestHeaders) {
+            const id = context?.requestHeaders?.upn || context?.requestHeaders?.requestingSystemId;
```
Comment (itaykl) [inline packages/polaris-typeorm/src/subscribers/common-model-subscriber.ts]: no need for ? inside the if

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 4]
```diff
@@ -12,7 +12,9 @@ export class DataVersionHandler {
         const extensions: PolarisExtensions =
             (manager.context && manager.context.returnedExtensions) || ({} as PolarisExtensions);
         connection.logger.log('log', 'Started data version job when inserting/updating entity');
-        const result = await this.selectDataVersionForUpdate(manager, connection);
+        const result = !extensions.dataVersion
```
Comment (ronkatz96) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts]: reverse not condition

File: packages/polaris-typeorm/src/subscribers/common-model-subscriber.ts (hunk 3 above)
Comment (ronkatz96) [inline packages/polaris-typeorm/src/subscribers/common-model-subscriber.ts]: check if you should switch:
context?.requestHeaders?.upn ?? context?.requestHeaders?.requestingSystemId;

File: packages/polaris-typeorm/src/subscribers/common-model-subscriber.ts [hunk 5]
```diff
@@ -114,4 +134,14 @@ export class CommonModelSubscriber implements EntitySubscriberInterface<CommonMo
...
+            if (entity.getLastUpdateTime() == null) {
+                entity.setCreatedBy(id);
+            }
```
Comment (ronkatz96) [inline packages/polaris-typeorm/src/subscribers/common-model-subscriber.ts:143]: check if still relevant:
if (entity.getLastUpdateTime() == null) {
                entity.setCreatedBy(id);
}

**Review by ronkatz96 — APPROVED** [review]

**Pull request merged**